### PR TITLE
Adds an element for noisy movement (wheelchairs, office chairs, trashcarts etc. etc.)

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Converts generic, copypasted behavior into an element.

## Why It's Good For The Game
I'd rather not have someone just copypaste the old thing, since there're a few more things to take into account now than just whether the item has gravity, and if that ever has to change, we will only have to modify one line than several.

## Changelog

:cl:
fix: Fixed the office chair being silent. My bad.
/:cl:

### CODE REVIEW

File: code/datums/elements/noisy_movement.dm [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	RegisterSignal(target, COMSIG_MOVABLE_MOVED, PROC_REF(play_sound))
+
+/datum/element/noisy_movement/Detach(datum/source)
+	UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+	return ..()
+
+/datum/element/noisy_movement/proc/play_sound(atom/movable/source, old_loc, movement_dir, forced)
+	SIGNAL_HANDLER
+	if(!forced && !CHECK_MOVE_LOOP_FLAGS(source, MOVEMENT_LOOP_OUTSIDE_CONTROL) && source.has_gravity())
+		playsound(src, movement_sound, volume, TRUE)
```
Comment (MrMelbert) [inline code/datums/elements/noisy_movement.dm:24]:
```suggestion
		playsound(source, movement_sound, volume, TRUE)
```
Hmm... tested code? 😉

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/elements/noisy_movement.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/datums/elements/noisy_movement.dm:24]: sus

File: code/datums/elements/noisy_movement.dm [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/datum/element/noisy_movement
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH_ON_HOST_DESTROY
```
Comment (LemonInTheDark) [inline code/datums/elements/noisy_movement.dm:2]: why does this element have `ELEMENT_DETACH_ON_HOST_DESTROY`, it only clears signals on destroy that happens automatically

**Comment by atlasle** [comment]
shouldn't this have the code improvement tag

File: code/datums/elements/noisy_movement.dm (hunk 1 above)
Comment (Ghommie) [reply code/datums/elements/noisy_movement.dm:24]: Ahah, you got me. I went to sleep without testing it. I will do it now though.

**Comment by Ghommie** [comment]
I can safely say it works now.

**Pull request merged**